week-4/C-more-complex-objects/exercise-3.js

```diff
@@ -31,13 +31,33 @@ var parkAvenueHouse = {
 */
 
 // returns the full name (first name + last name) of the owner of the house
-function getOwnerFullName(house) {}
+function getOwnerFullName(house) {
+  return house.currentOwner.firstName+ ' ' + house.currentOwner.lastName;
+}
 
 // returns an array of the owners' email addresses of the two houses
-function getEmailAddresses(house1, house2) {}
+function getEmailAddresses(house1, house2) {
+  var arr = [];
+  //arr[0] = house1.currentOwner.email; 
+  //arr[1] = house2.currentOwner.email;
+  arr.push(house1.currentOwner.email);//at the end of the array
+  arr.push(house2.currentOwner.email);
+  return arr;
+}
 
 // returns the address for the cheapest house out of the two
-function getCheapestAddress(house1, house2) {}
+function getCheapestAddress(house1, house2) {
+  var houseOnePrice = house1.price;
+  var houseTwoPrice = house2.price;
+
+  if(houseOnePrice < houseTwoPrice){
+    return house1.address;
+  } else if (houseOnePrice > houseTwoPrice){
+    return house2.address;
+  }else{
+    return 'price are the same';

```

Comment:
Be careful here. If the caller of getCheapestAddress expects to get a house address back and they get this text instead, it might be a bit of a shock if they try to do anything with it (for instance, look up where it is in a list).

It's safe to return either address here, both are equally cheap.